homeassistant/components/mailbox/__init__.py

```diff
@@ -86,6 +87,8 @@ async def async_setup_platform(
         component = EntityComponent[MailboxEntity](
             logging.getLogger(__name__), DOMAIN, hass, SCAN_INTERVAL
         )
+        # pylint: disable-next=[protected-access]
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, component._async_shutdown)
```

Comment:
I am not sure why  `component.async_setup` is not called on `mailbox` platform.
This at least ensure that it is shutdown correctly on stop.